backend/app/Services/Warehouse/WarehouseStockService.php

```
@@ -0,0 +1,106 @@
+    public function adjust(array $keys, float $signed, ?string $unit = null): WarehouseStock
+    {
+        $keys = $this->normalizeKeys($keys);
+
+        $stock = WarehouseStock::query()->where($keys)->lockForUpdate()->first();
+
+        if (! $stock) {
+            // lockForUpdate() can only lock rows that exist, so two concurrent callers
+            // can both miss and then race to insert. The partial unique index decides
+            // the winner; the loser re-reads the row it lost to instead of failing.
+            try {
+                $stock = WarehouseStock::create([...$keys, 'quantity' => 0, 'unit_of_measure' => $unit]);
```

PHPStan (2.2.8)

[error] Call to an undefined static method App\Models\WarehouseStock::create().


(staticMethod.notFound)